Zombie child evolution expansion.

Rivet-the-Zombie opened the pull request:
Poor zombie kids didn't used to have any future prospects to look forward to as the world becomes more and more mutated over time.

This PR adds six new evolutionary possibilities for zombie kids. Fairly fragile on their own with reduced hitpoints compared to vanilla zombie children, each trades off various other traits for special attacks, skills, and abilities that change their purpose and tactics.

Shrieklings do what kids do best: run around fast and scream a lot. Very fragile.

Zombie waifs are skeletal (read as: hard to hit with ranged attacks) zombie children that can scream to draw attention.

Snotgobblers are mini grabber zeds that bleed acid when they die like a spitter.

Anklebiters are quick little things with enhanced nibbling capability. Too badly mutated to look human any more, they no longer evoke guilt when killed.

Sproglodytes are tiny night hunters. Nearly blind in the daylight, at night they can hunt you down using their night vision, pouncing upon you from far away.

Creepy crawlers are also specialized for night operations, but they prefer to scream and draw in other zeds when they see you. Too badly mutated, they too no longer cause you to feel guilt when killed.

Cyrano7 commented:
Now i'll think twice about walking past the caged-in play ground full of little zombie children.

Skidborg commented:
This sounds horrifying. I'm looking forward to it.

hitbutton commented:
A lot of this seems cool, but there are quite a few aspects that don't seem to match the general tone/design.

Zombie names are generally
a) functional - you can broadly guess characteristics/special attacks even if you haven't met one before
b) serious - the game does have a lot of comedy aspects, but not generally as overt as common enemies being called "snotgobblers"

Zombies are generally pretty focused. They don't tend to have multiple special abilities unless they're thematically consistent.
All the children have smallness as a characteristic, making them evasive. Adding an extra ability on top of that seems reasonable, but adding two unrelated ones (acid + grab for snotgobbler, skeletal + scream for waifs) seems a bit incoherent.

Skidborg commented:
I'd see the combined abilities as more of an omission on the part of regular zombies than an overreach here.  Evolution trees would benefit from being fleshed out at every level.

Lain- commented:
Pretty cool expansion for the zombie children, though I share some of @hitbutton's concerns about the special abilities. Zombie waifs (scream + skeletal) has the potential to be very frustrating to fight again, like how zombie dogs was annoying several versions ago.

I think each variant should only have one ability and a defining characteristic. Screaming variant, crawling grabbing variant and night fighting variant? Every variant should inflict no guilt when killed as they are too mutated and misshapened.

hitbutton commented:
> I'd see the combined abilities as more of an omission on the part of regular zombies than an overreach here. Evolution trees would benefit from being fleshed out at every level.

Its not so much that I don't think zombies should have multiple abilities, but those abilities should be thematically consistent with a single concept
.
Reading the description for snotgobbler, I can see the rationale for grabbing and acid. I think I'd have less of a problem with that combination if they were called something more dull and descriptive like "sticky child zombie".

hitbutton commented on data/json/monstergroups.json:
```diff
@@ -1293,6 +1293,19 @@
   },{
     "type":"monstergroup",
+    "name" : "GROUP_CHILD_ZOMBIE_UPGRADE",
+    "default" : "mon_zombie_spitter",
```
The default for the mongroup probably shouldn't be mon_zombie_spitter

LlubNek commented:
>I think I'd have less of a problem with that combination if they were called something more dull and descriptive like "sticky child zombie".

"dull" is not a good thing in a game, and not really realistic either.  People don't use dull names for their opponents in combat.  If these creatures were named by survivors or soldiers in the field, they'd have pretty colourful names.

If you want dull descriptive names, maybe you should add some item or file in the labs which adds scientific names, lab notes and observations for various monsters.

hitbutton commented:
"dull" is a good thing in a game where it allows you to intuit information directly from a name, rather than say "what the hell is a snotgobbler?"

LlubNek commented:
> "dull" is a good thing in a game where it allows you to intuit information directly from a name, rather than say "what the hell is a snotgobbler?"

"what the hell is a snotgobbler?" is answered by the description.  Just mouse over it or use shift-V and you'll find out.  A question like that isn't a bad thing anyway:  it caches the player's attention and gives them something new to discover.

LlubNek commented:
> The default for the mongroup probably shouldn't be mon_zombie_spitter
Monster groups are used when spawning monsters.  If a mob should spawn with child zombies, it should be in a child zombie group.  You'd put them in mon_zombie_spitter if you wanted them to spawn like spitter zombies (which you probably don't).

Coolthulhu commented on data/json/monsters.json:
```diff
@@ -2901,12 +2901,238 @@
+    },{
+    "type" : "MONSTER",
+    "id" : "mon_zombie_sproglodyte",
+    "name": "sproglodyte",
```
That name is kinda weird...
It leaps, cave-related name doesn't fit here.

Maybe some vampire reference thing?
Chinese vampires (jiangshi) move by hopping.
"Child of the night"?
Maybe something about bats?

Coolthulhu commented on data/json/monsters.json:
```diff
@@ -2901,12 +2901,238 @@
+    "special_attacks":[["LEAP", 10]],
+    "description":"This crouching child-mutant's face is dominated by a pair of huge black eyes and its fingertips end in sharp claws.",
+    "//": "GUILT because it still looks enough like a child to evoke pity",
+    "flags":["SEES", "HEARS", "SMELLS", "STUMBLES", "WARM", "BLEED", "GUILT", "BASHES", "POISON", "NO_BREATHE", "REVIVES", "BONES"]
```
"GUILT" with leaping and high speed could lead to it being annoying.
Guilt works best when you have a choice to leave things but choose to kill them anyway.

Coolthulhu commented on data/json/monsters.json:
```diff
@@ -2901,12 +2901,238 @@
+        ]
+    },
+    "hp":25,
+    "death_function":["ACID", "NORMAL"],
```
ACID death doesn't do much. Boomer death maybe?

Coolthulhu commented on data/json/monsters.json:
```diff
@@ -2901,12 +2901,238 @@
+    "special_attacks":[["SHRIEK", 5]],
+    "description":"What was once a child is now a mutant beast with blackened skin and massive eyes.  This abomination's vile form makes mockery of its human origin.",
+    "//": "no GUILT because it no longer looks enough like a child to evoke pity",
+    "flags":["SEES", "HEARS", "SMELLS", "STUMBLES", "WARM", "BLEED", "BASHES", "POISON", "NO_BREATHE", "REVIVES", "BONES"]
```
I assume it crawls on all fours rather than like crawling zed (just on arms).
Could get climbing, to make it a bit more insect-like.
Almost all of them shriek, maybe move the role of "night shrieker" to the waif one (it doesn't differ from shriekling much) and give this one, say, acid barf (the weak acid zed one, not spitter or corrosive one)?

The description could also mention it runs on all fours, currently only name and speed suggest that.

chaosvolt commented:
Hmm. Interesting concepts. I can see these being a pain to sprite. o3o

Coolthulhu commented:
Good idea to flesh out the zed children. Now only human zed who doesn't evolve is the crawler (and it could just evolve the leg control back or something).
It will make parks a lot louder.

All of those are rather weak, so their evolution could happen sooner than adult ones. This would also fit the whole child themes, because growing up.

> Zombie names are generally
> a) functional - you can broadly guess characteristics/special attacks even if you haven't met one before
> b) not silly - the game does have a lot of comedy aspects, but not generally as overt as common enemies being called "snotgobblers"

This
Creepy crawler, shriekling and anklebiter are descriptive - the name is enough to "see" the zed.

Waif describes only the skeletal part.
"Banshee" would capture both skeletal and screaming, but would be a bit too "dangerous" and also suggest an adult zed.

Snotgobbler only suggests that it is disgusting, not in what way. It only suggests what will happen when you kill it, which isn't much, because ACID death is very weak (it is useful for the player, as other zeds can be lead into it).
I'd change that acid death to boomer death (which would make it meaningful).
Not sure about the name. Bilebag (a bit too... basic)? Clinger (but it doesn't climb)? ...boomster?

Sproglodyte (sounds like troglodyte spriggan) sounds like something that likes darkness (30 night vision), but its important feature (leaping) isn't really conveyed.
Something bat-related would fit the best, I think.

chaosvolt commented:
Hmm. Kinder der Nacht maybe? o3o

Also...regarding that mention of crawlers, it's aslo supposed to allow for undead who are implied to have been bisected, right? Though fleshing out evolved crawlers into leggy and legless variants will allow for that.

Anyway, back on topic. If we had a nastier variant of acid death...boomer would work though.

hitbutton commented:
My preference would be <descriptive adjective> zombie child 
Batlike zombie child, Bilious zombie child etc

Rivet-the-Zombie commented:
I'll address these issue this evening.

mugling commented:
> Bilebag

Sounds ok

Rivet-the-Zombie commented:
Fixed default upgrade type.
Restored BOOMER death effect to the snotgobbler (this was its original intended behavior).
Renamed 'zombie waif' to 'howling waif' to reference its ability.
Removed GUILT from sproglodyte.

>Could get climbing, to make it a bit more insect-like.

It would already have it if the game supported such an ability.

>Zombie names are generally
a) functional - you can broadly guess characteristics/special attacks even if you haven't met one before

We've already got stuff like 'boomer' and 'spitter' which really only make much sense if you've played the L4D games - yeah it's vaguely descriptive, but unless I'd already played those games I'd be confused as to why 'spitters' vomit acid as opposed to, you know, actually spitting on people.

>Sproglodyte (sounds like troglodyte spriggan) sounds like something that likes darkness (30 night vision), but its important feature (leaping) isn't really conveyed.

It's a dual-meaning usage: 'troglodyte  as in cave-dweller, and also as in 'Pan troglodytes' - which covers their hoppy behavior.

Coolthulhu commented:
> It would already have it if the game supported such an ability.

There is the "CLIMBS" flag. It doesn't do much yet in 2D mode (lets it walk over wire fences and pits), but allows it to climb up and down z-levels in 3D mode.

> We've already got stuff like 'boomer' and 'spitter' which really only make much sense if you've played the L4D games

Spitter spits something harmful, boomer explodes. Out of the L4D zeds, it's the hunter which is the least descriptive (jumping on stuff isn't really "hunting").
All of them have an advantage over creative names though: they sound like a name people in the game gave them rather than something totally arbitrary. Simple names that can be shouted to warn other survivors.

Out of human zeds, the lowest "standard of naming" is probably the bloated one - it describes only how it looks, not what it does. The second would be corrosive and/or acidic - they're both non-natural ("look out, a corrosive!" doesn't sound good) and only suggest that they are related to acid, not how.
Those three could use a rename to something less technical and more "survivor-y".

chaosvolt commented:
> Out of human zeds, the lowest "standard of naming" is probably the bloated one - it describes only how it looks, not what it does.

It does have "gasbag" in its ID though. Gasbag zombie sounds a bit silly, but would be more descriptive. Has it always been bloated zombie, or did it get renamed?

> ("look out, a corrosive!" doesn't sound good)

"Look out, a fat!" Problem is in name usage as well, for all the adjective-based zeds. XP

"Corrosive zombie" is still vague and a bit of a mouthful, admittedly. Adicic zombie is at least less evocative in a way that adequately matches up with it being less of a threat, whereas corrosive does sound more menacing (subjective though) than spitter. So at least the idea they're more of a threat can be conveyed.

Anyway, to re-on-topic-ify us, most of the names seem reasonable now, though snotgobbler and sproglodyte are a bit odd. Issue here is I'm unsure what else to suggest for alternatives. Mini-boomer and springy sprout? o3o

Malkeus commented:
Poppers and Leaplings. Poppers evokes one of the horrors of teendom, acne, the description could describe them as having pustules on their bloated flesh, like some kind of acne covered, pus filled mess. That grossed me out just to write.
Leaplings not happy with...but it was the first thing to come to mind. I first thought of zerglings, but that's got a lot of imagery associated with it already. Bouncing Bettys? Those will cut you off at the knees...

chaosvolt commented:
Whee, gonna have to hurry up with the spriting then. o3o
